Source/Core/Core/PrimeHack/Mods/FpsControls.cpp

```diff
@@ -136,17 +145,29 @@ namespace prime {
     handle_beam_visor_switch(prime_one_beams, prime_one_visors);
 
     // Allows freelook in grapple, otherwise we are orbiting (locked on) to something
-    if (read32(mp1_static.orbit_state_address) != ORBIT_STATE_GRAPPLE &&
-      read8(mp1_static.lockon_address)) {
+    bool locked = (read32(mp1_static.orbit_state_address) != ORBIT_STATE_GRAPPLE &&
+                   read8(mp1_static.lockon_address));
+    
+    if (locked) {
       write32(0, mp1_static.yaw_vel_address);
-      return;
+      // Calculate the pitch based on the XF matrix of player so that we always
+      // write out the pitch even while locked onto the target, this prevents
+      // the view from snapping back after releasing a lock-on
```

Comment:
I would place this comment before the `calculate_pitch_locked` function is defined, rather than in the run function, to try keep things clean.